### ISSUE TITLE
User-Defined Activation Events for Extensions

### ISSUE THREAD

**Issue by mattorp** [microsoft/vscode#233187]
Allow users to set custom activation events in settings.json for each extension, independent of the extension’s package.json. For instance:

```
"extensions.activationOverrides": {
    "git-graph": {
        "activationEvents": ["onCommand:git-graph.view:showGraph"]
    }
}
```

This would allow users to define custom activation events for VSCode extensions. Currently, the activation of an extension is determined by predefined events specified in the extension’s package.json (such as onStartupFinished, onCommand:<command>, etc.). However, there are many cases where a user might want more control over these events to prevent certain extensions from running until explicitly needed.

For example, I have extensions like [Git Graph](https://github.com/mhutchie/vscode-git-graph), which I use occasionally, but not enough to justify having it always active. I prefer it to activate only when I run a specific command, such as “Show Graph.” This would reduce resource usage and streamline the extension experience in VSCode.

**Comment by joyceerhl** [comment]
@isidorn have we reached out to Git Graph re: their `*` activation usage?

**Comment by isidorn** [comment]
No. Their last commit is from 3 years ago. But let me send them an email and cc you
